packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -121,58 +121,73 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
   });
 
   const displayVisitStopDateTimeFields = useMemo(
-    () => visitToEdit?.stopDatetime || showVisitEndDateTimeFields,
+    () => Boolean(visitToEdit?.stopDatetime || showVisitEndDateTimeFields),
     [visitToEdit?.stopDatetime, showVisitEndDateTimeFields],
   );
 
   const visitFormSchema = useMemo(() => {
+    const createVisitAttributeSchema = (required: boolean) =>
+      required
+        ? z
+            .string({
+              required_error: t('fieldRequired', 'This field is required'),
+            })
+            .refine((value) => !!value, t('fieldRequired', 'This field is required'))
```

Comment:
Why are we using 2 checks for the same thing?

Reply:
Oops!